File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        r = [1]*n

        for i in range(m-1):
            newR = [1]*n
            for j in range(n-2, -1, -1):
                newR[j] = newR[j+1] + r[j]
            r = newR
        
        return r[0]

File: 0062-unique-paths/0062-unique-paths.kt
class Solution {
    fun uniquePaths(m: Int, n: Int): Int {
        var row: Array<Int> = Array(n) { 1 }

        for (i in 0 until m-1) {
            var newRow: Array<Int> = Array(n) { 1 }
            for (j in n - 2 downTo 0) {
                newRow[j] = newRow[j+1] + row[j]
            }
            row = newRow
        }

        return row[0]
    }
}